### PR TITLE
adds power creep

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: code/game/objects/items/weapons/power_cells.dm [hunk 1]
```diff
@@ -112,6 +112,27 @@
...
+	charge = 12000
+	w_type = RECYK_BIOLOGICAL
+	minor_fault = 1
+
+/obj/item/weapon/cell/crepe/attack_self(var/mob/living/user)
+	user.visible_message("<span class = 'notice'>\The [user] takes a bite out of \the [src]</span>", "<span class = 'warning'>You take a bite out of \the [src]</span>")
+	spawn(rand(1,3) SECONDS)
+		var/power_to_use = min(charge, rand(800,1200))
+		playsound(loc, 'sound/effects/eleczap.ogg', 80, 1)
+		use(power_to_use)
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/power_cells.dm]: The sound effect and the rest of the code should be indented under an `if` for this - if the cell can't give you the power you want, this returns a 0.

**Review by ComicIronic — DISMISSED** [review]

File: code/game/objects/items/weapons/power_cells.dm [hunk 2]
```diff
@@ -112,6 +112,27 @@
 	w_type = RECYK_BIOLOGICAL
 	minor_fault = 1
 
+/obj/item/weapon/cell/crepe
+	name = "power crepe"
```
Comment (PJB3005) [inline code/game/objects/items/weapons/power_cells.dm]: crêpe

File: code/game/objects/items/weapons/power_cells.dm [hunk 3]
```diff
@@ -112,6 +112,27 @@
...
+	desc = "Warning: May contain dairy products, 12,000kw of searing death, gluten."
```
Comment (PJB3005) [inline code/game/objects/items/weapons/power_cells.dm]: `kW`

File: code/modules/food/recipes_microwave.dm [hunk 4]
```diff
@@ -2272,4 +2272,15 @@
...
+	
+/datum/recipe/power_crepe
+	reagents = list(RADIUM = 5)
+	items = list(
+		/obj/item/weapon/cell,
+		/obj/item/weapon/reagent_containers/food/snacks/dough,
+		/obj/item/stack/cable_coil,
+		)
+	result = /obj/item/weapon/cell/crepe
+	time = 300
```
Comment (Kurfursten) [inline code/modules/food/recipes_microwave.dm:2285]: since no other kitchen recipe uses this, it might confuse people

Comment (ghost) [reply code/modules/food/recipes_microwave.dm:2285]: It's for balancing

**Comment by FudgePucker** [comment]
Good shit :ok_hand:

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/power_cells.dm [hunk 5]
```diff
@@ -112,6 +112,27 @@
...
+
+/obj/item/weapon/cell/crepe/attack_self(var/mob/living/user)
+	user.visible_message("<span class = 'notice'>\The [user] takes a bite out of \the [src]</span>", "<span class = 'warning'>You take a bite out of \the [src]</span>")
+	spawn(rand(1,3) SECONDS)
+		var/power_to_use = min(charge, rand(800,1200))
+		playsound(loc, 'sound/effects/eleczap.ogg', 80, 1)
+		if(use(power_to_use))
+			user.adjustFireLoss(power_to_use/100) //So 8 to 12 damage
+			user.visible_message("<span class = 'notice'>\The [user] is electrocuted by \the [src]</span>", "<span class = 'warning'>You are [pick("frazzled","electrocuted","zapped")] by \the [src]!</span>")
+			user.set_light(2,2,"#ffff00")
```
Comment (PJB3005) [inline code/game/objects/items/weapons/power_cells.dm]: Doesn't this fuck with things like getting set on fire?

File: code/game/objects/items/weapons/power_cells.dm (hunk 2 above)
Comment (ghost) [reply code/game/objects/items/weapons/power_cells.dm]: Don't have atom on this computer, will have to wait till monday for accents

File: code/game/objects/items/weapons/power_cells.dm (hunk 5 above)
Comment (ghost) [reply code/game/objects/items/weapons/power_cells.dm]: If you're on fire, why are you eating something that deals burn damage?

File: code/game/objects/items/weapons/power_cells.dm (hunk 2 above)
Comment (Shadowmech88) [reply code/game/objects/items/weapons/power_cells.dm]: What if you add it via editing the file on Github?

File: code/game/objects/items/weapons/power_cells.dm (hunk 5 above)
Comment (ComicIronic) [reply code/game/objects/items/weapons/power_cells.dm]: Bzzt, wrong answer.

Comment (ghost) [reply code/game/objects/items/weapons/power_cells.dm]: Well, how would you recommend doing it?

Most things that update lighting update it every so few ticks (The light virus effect, radglow, and I would assume fire), but there's no real way to revert back to that with this.

Two ways I can think of doing it. Store the light settings in temporary variables to be restored after the cooldown, or check if the person is alight already and if so don't light them up.

**Comment by Sarcalogo** [comment]
deus ex human revolution prequel
https://www.youtube.com/watch?v=lY5hDzWfnxw

**Comment by ghost** [comment]
???

**Comment by Sarcalogo** [comment]
*bing bing yahoo*

image

File: code/game/objects/items/weapons/power_cells.dm (hunk 5 above)
Comment (PJB3005) [reply code/game/objects/items/weapons/power_cells.dm]: To handle this properly you need wrappers, managers for lighting colors and whatever.

so... I'd just recommend it not doing anything if they're already lit. Not perfect but least effort.

What you'd need is a list of light effects on mobs, then produce a final color, range and power based on all of those.

File: code/game/objects/items/weapons/power_cells.dm (hunk 3 above)
Comment (Exxion) [reply code/game/objects/items/weapons/power_cells.dm]: That's still the wrong unit, so who cares

Comment (PJB3005) [reply code/game/objects/items/weapons/power_cells.dm]: Oh yeah.

`kJ`

Comment (ghost) [reply code/game/objects/items/weapons/power_cells.dm]: wouldn't it be just (J) Joules in this case? or 12kJ ?

**Review by PJB3005 — APPROVED** [review]

**Comment by Megamaw** [comment]
are we ready to approve because this is an amazing pr

**Comment by Intigracy** [comment]
Make sure you update the wiki.

**Pull request merged**